extensions/mapstore/src/main/java/com/hazelcast/mapstore/MappingClosingIterator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.mapstore;
+
+import java.util.Iterator;
+import java.util.function.Function;
+
+/**
+ * Iterator providing mapping for the iterated objects from type T to R and
+ * also implementing {@link AutoCloseable}.
```

Comment:
Why implementing `AutoCloseable`? Where do we close it?

Reply:
The map store doesn't call it, it's closed when `loadAllKeys` is finished here https://github.com/hazelcast/hazelcast/blob/14ef0dbefa4a6fa00a50a8873aeea02ec6c3f5e4/hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java#L478

Fixed `AutoCloseable` -> `Closeable`
ebee23f5